src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,85 @@
+package duke.task;
+
+import duke.exception.DukeException;
+import duke.exception.DukeExceptionType;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private static final int MAXSIZE = 100;
+
+    private ArrayList<Task> taskList;
+
+
+    public TaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    public TaskList(String[] tasks) {
+        taskList = new ArrayList<>();
+        loadTaskList(tasks);
+    }
+
+    public void addTask(Task task) throws DukeException {
+        if (taskList.size() == MAXSIZE) {
+            throw new DukeException(DukeExceptionType.FULL_TASK_LIST);
+        }
+        taskList.add(task);
+    }
+
+    private void loadTaskList(String[] tasks) {
+        for (String task : tasks) {
+            loadTask(task);
+        }
+    }
+
+    private void loadTask(String task) {
+        try {
+            switch (task.charAt(0)) {
+            case 'T':
```

Comment:
Ah! in most other places you have constants in place of magic literals. Why not here?